src/dune_rules/pkg_rules.ml

```diff
@@ -195,6 +186,48 @@ module Install_cookie = struct
     | Some f -> f
 end
 
+module Env_update = struct
+  include Dune_lang.Action.Env_update
+
+  let update kind ~new_v ~old_v ~f =
+    if new_v = "" then old_v
+    else
+      match kind with
+      | `Colon ->
+        let old_v = Option.value ~default:[] old_v in
+        Some (f ~old_v ~new_v)
+      | `Plus -> (
+        match old_v with
+        | None | Some [] -> Some [ Value.String new_v ]
+        | Some old_v -> Some (f ~old_v ~new_v))
+
+  let append = update ~f:(fun ~old_v ~new_v -> old_v @ [ Value.String new_v ])
```

Comment:
Maybe it would make sense not to store the env as a list but as a sort of deque and only materialize the list in the end?

OTOH, it probably doesn't matter unless people append large amounts of variables.

Reply:
Yeah, it probably won't matter. If this is going to be optimized, I'd rather let the opam maintainers do it and give us some generic component we can replace this code with.